initializer/mount.go

```diff
@@ -0,0 +1,246 @@
+		Therefore we expose the defined PVC as a block VolumeDevice to our
+		initializer container. This allows the initializer to setup the
+		encryption on the block device located at [device-path] using cryptsetup,
+		the encryption passphrase is derived from the UUID of the LUKS formatted
+		block device and the current workload secret.
+
+		The mapped decrypted block device can then be shared with other containers
+		on the pod by setting up a shared VolumeMount on the specified [mount-point],
+		where the mapper device will be mounted to.`,
+		RunE: setupEncryptedMount,
+	}
+	cmd.Flags().StringP("device-path", "d", "/dev/csi0", "path to the volume device to be encrypted")
+	cmd.Flags().StringP("mount-point", "m", "/state", "mount point of decrypted mapper device")
+	must(cmd.MarkFlagRequired("device-path"))
+	must(cmd.MarkFlagRequired("mount-point"))
+
+	return cmd
+}
+
+// parseSetupEncryptedMountFlags returns a luksVolume, representing the provided flags to the subcommand, which is then used to setup the encrypted volume mount.
+func parseSetupEncryptedMountFlags(cmd *cobra.Command) (*luksVolume, error) {
+	devicePath, err := cmd.Flags().GetString("device-path")
+	if err != nil {
+		return nil, err
+	}
+	mountPoint, err := cmd.Flags().GetString("mount-point")
+	if err != nil {
+		return nil, err
+	}
+	hash := md5.Sum([]byte(devicePath + mountPoint))
+	mappingName := hex.EncodeToString(hash[:8])
+	return &luksVolume{
+		devicePath:       devicePath,
+		mappingName:      mappingName,
+		volumeMountPoint: mountPoint,
+	}, nil
+}
+
+// setupEncryptedMount sets up an encrypted LUKS volume mount using the device path and mount point, provided to the setupEncryptedMount subcommand.
+func setupEncryptedMount(cmd *cobra.Command, _ []string) error {
+	logger, err := logger.Default()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: creating logger: %v\n", err)
+		return err
+	}
+	luksVolume, err := parseSetupEncryptedMountFlags(cmd)
+	if err != nil {
+		return fmt.Errorf("parsing setupEncryptedMount flags: %w", err)
+	}
+	ctx := cmd.Context()
+	if !isLuks(ctx, logger, luksVolume.devicePath) {
+		// TODO(jmxnzo) check what happens if container is terminated in between formatting
+		if err := luksFormat(ctx, luksVolume.devicePath, workloadSecretPath); err != nil {
+			return err
+		}
+	}
+	if err := openEncryptedDevice(ctx, luksVolume); err != nil {
+		return err
+	}
+	// The decrypted devices with <name> will always be mapped to /dev/mapper/<name> by default.
+	if err := setupMount(ctx, logger, "/dev/mapper/"+luksVolume.mappingName, luksVolume.volumeMountPoint); err != nil {
+		return err
+	}
+
+	if err := os.WriteFile("/done", []byte(""), 0o644); err != nil {
+		return fmt.Errorf("Creating startup probe done directory:%w", err)
+	}
+	// Waits for SIGTERM signal and then properly release the resources.
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
I think we should register the channel at least at the top of this function as to not miss any signals. We could also even decide to cancel the context then, but this can be done in a follow-up.